src/xdsl/rewriter.py

```diff
@@ -117,8 +119,9 @@ class OperandUpdater:
     Provides functionality to bookkeep changed results and to access and update them.
     """
 
-    result_mapping: Dict[SSAValue, SSAValue] = field(default_factory=dict)
-
+    result_mapping: Dict[SSAValue,
+                         Optional[SSAValue]] = field(default_factory=dict)
```

Comment:
Here we have a mapping to potentially `None` values, so we should later reflect that potential `None` in the types that use this mapping.